index.js

```diff
@@ -195,7 +202,22 @@ class Replicate {
       throw new Error(`Prediction failed: ${prediction.error}`);
     }
 
-    return prediction.output;
+    if (!strict) {
+      return prediction.output;
+    }
+
+    const response = await this.models.versions.get(identifier.owner, identifier.name, identifier.version);
+    const { openapi_schema: schema } = response;
```

Comment:
I think fetching the schema in a separate request is a deal-breaker here. That's something the Python client has been [doing](https://github.com/replicate/replicate-python/blob/ecfedfb8c645f42b60fc318948a8b222d8bf2869/replicate/run.py#L53) from the start, and it's added a lot of unnecessary overhead.

It's not as magical, but could we instead implement coerced `run` on a model / version? That way, you'd avoid duplicate requests and have a clearer expectations around how/when schemas are validated.

Reply:
> It's not as magical, but could we instead implement coerced run on a model / version? That way, you'd avoid duplicate requests and have a clearer expectations around how/when schemas are validated.

I'm not sure what you mean here?